docs/eventing/sources/pingsource.md

```diff
@@ -1,196 +1,261 @@
 
-This example shows how to send an event every minute to a Event Display Service.
+### Producers
 
-### Creating a namespace
+Producers create events. Using GitHub as an example, the producer is the GitHub webhook that sends
+information when a pull request is created.
+Essentially, you ask the producer for events that they are capable of sending and you tell the
+producer where to send those events.
 
-Create a new namespace called `pingsource-example` by entering the following
-command:
+Producers send the event data per specifications that make it easy for consumers to handle it.
+Knative uses the CloudEvents specifications by default.
+For more information, see the [Cloud Events](https://cloudevents.io/) website.
 
-```shell
-kubectl create namespace pingsource-example
-```
+### Consumers
+
+Consumers consume the data that producers create. Consumers are usually functions.
 
-### Creating the Event Display Service
+In the GitHub example above, a consumer might update a ticket in the ticketing system with the data
+from the pull request.
 
-In this step, you create one event consumer, `event-display` to verify that
-`PingSource` is properly working.
 
-To deploy the `event-display` consumer to your cluster, run the following
-command:
+## Create a consumer (Knative service)
+
+To create a consumer, and verify that `PingSource` is working, create a simple Knative service that
+receives CloudEvents and writes them to stdout.
+
+{{< tabs name="create-service" default="YAML" >}}
+{{% tab name="YAML" %}}
+Create the service from stdin by running:
 
 ```shell
-kubectl -n pingsource-example apply -f - << EOF
-apiVersion: apps/v1
-kind: Deployment
+cat <<EOF | kubectl create -f -
+apiVersion: serving.knative.dev/v1
+kind: Service
 metadata:
   name: event-display
 spec:
-  replicas: 1
-  selector:
-    matchLabels: &labels
-      app: event-display
   template:
-    metadata:
-      labels: *labels
     spec:
       containers:
-        - name: event-display
-          image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
+        - image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
+EOF
+```
+{{< /tab >}}
 
----
+{{% tab name="kn" %}}
+Create the service using the kn command-line interface by running:
 
-kind: Service
-apiVersion: v1
-metadata:
-  name: event-display
-spec:
-  selector:
-    app: event-display
-  ports:
-  - protocol: TCP
-    port: 80
-    targetPort: 8080
-EOF
+```shell
+kn service create event-display --image gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
 ```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Create a producer (PingSource)
 
-### Creating the PingSource
+Create a producer (PingSource) that sends events to your consumer every two minutes.
+The `sink` element in the metadata describes where to send events.
+In this case, events are sent to a service with the name `event-display`.
+This is a tight connection between the producer and consumer.
```

Comment:
Again, remove parentheses. Feels over-explained.
I don't think these all need to be separate lines either, it could just be a paragraph? The flow just feels off here IMO.

I think in general this includes too much context between steps of a procedure - IMO all the conceptual info should go in a generic event sources concept doc and the steps should be in a simple procedural topic.
Also, rather than explaining things like `sink` here, can we use the callout workaround style that was established; https://docs.google.com/document/d/1tigolRzKhronvgoAJfszE-XHWWd0LQ8o_j0D73O9KvM/edit#heading=h.ka7avn97fckt

Reply:
Okay, I'll cut back a bit.